File: 240412/격자 위의 편안한 상태/comfortable-state-on-the-grid.py
N, M = map(int, input().split())
board = [[0] * N for _ in range(N)]
chk_dirs = [(-1, 0), (1, 0), (0, -1), (0, 1)]

for _ in range(M):
    r, c = map(lambda x: int(x) - 1, input().split())
    board[r][c] = 1

    colors = 0
    for dr, dc in chk_dirs:
        mr, mc = r + dr, c + dc
        if 0 <= mr < N and 0 <= mc < N and board[mr][mc] == 1:
            colors += 1

    print(1 if colors == 3 else 0)

File: 260205/격자 위의 편안한 상태/comfortable-state-on-the-grid.cpp
#include <bits/stdc++.h>
using namespace std;

int main() 
{
    int N, M, r, c;
    cin >> N >> M;
    vector<vector<int>> board(N, vector<int>(N, 0));
    const int dr[4] = {-1, 1, 0, 0};
    const int dc[4] = {0, 0, -1, 1};

    for (int i = 0; i < M; i++) 
    {
        cin >> r >> c;
        r -= 1; c -= 1;
        board[r][c] = 1;

        int colors = 0;
        for(int k = 0; k < 4; k++)
        {
            int mr = r + dr[k];
            int mc = c + dc[k];
            if(0 <= mr && mr < N && 0 <= mc && mc < N && board[mr][mc] == 1)
                colors += 1;
        }

        cout << (colors == 3 ? 1 : 0) << endl;
    }
    return 0;
}
